astropy/units/format/console.py

```diff
@@ -68,30 +69,51 @@ def to_string(cls, unit):
                 s = cls.format_exponential_notation(unit.scale)
 
             if len(unit.bases):
-                positives, negatives = utils.get_grouped_by_powers(
-                    unit.bases, unit.powers
-                )
-                if len(negatives):
-                    if len(positives):
-                        positives = cls._format_unit_list(positives)
+                if cls.inline:
```

Comment:
Wouldn't it make more sense to just overwrite the definition of `to_string()` in the definition of `ConsoleInline`?

Actually, possibly best is to make `ConsoleInline` the top class, and let `Console` inherit from it, since then it can use `super()` for the case that there is no `denominator`.